src/tests/Interop/ArrayMarshalling/StructArray/MarshalStructArrayNative.cpp

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#include <xplatform.h>
+#include <platformdefines.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+extern "C" DLL_EXPORT void STDMETHODCALLTYPE ForwardToCallback(void* value, void(STDMETHODCALLTYPE* callback)(void*))
```

Comment:
Would `Marshal.DestroyStructucture` hit the buggy path? `Marshal.DestroyStructucture` may be simpler and more direct way to test it.

Reply:
I'll check and see if I can do it with that instead. That would be significantly cleaner.